system/nxcamera/nxcamera_main.c

```diff
@@ -0,0 +1,564 @@
+
+#ifdef CONFIG_NXCAMERA_INCLUDE_HELP
+static int nxcamera_cmd_help(FAR struct nxcamera_s *plooper, char *parg);
+#endif
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const struct mp_cmd_s g_nxcamera_cmds[] =
+{
+#ifdef CONFIG_NXCAMERA_INCLUDE_PREFERRED_DEVICE
+  {
+    "input",
+    "videodev",
+    nxcamera_cmd_input,
+    NXCAMERA_HELP_TEXT("Specify a preferred capture device")
+  },
+#endif
+  {
+    "output",
+    "fbdev|filename",
+    nxcamera_cmd_output,
+    NXCAMERA_HELP_TEXT("Specify a display device or filename")
+  },
+#ifdef CONFIG_NXCAMERA_INCLUDE_HELP
+  {
+    "h",
+    "",
+    nxcamera_cmd_help,
+    NXCAMERA_HELP_TEXT("Display help for commands")
+  },
+  {
+    "help",
+    "",
+    nxcamera_cmd_help,
+    NXCAMERA_HELP_TEXT("Display help for commands")
+  },
+#endif
+  {
+    "stream",
+    "width height framerate format",
+    nxcamera_cmd_stream,
+    NXCAMERA_HELP_TEXT("Video stream test (format is fourcc)")
+  },
+  {
+    "stop",
+    "",
+    nxcamera_cmd_stop,
+    NXCAMERA_HELP_TEXT("Stop stream")
+  },
+  {
+    "q",
+    "",
+    nxcamera_cmd_quit,
+    NXCAMERA_HELP_TEXT("Exit NxCamera")
+  },
+  {
+    "quit",
+    "",
+    nxcamera_cmd_quit,
+    NXCAMERA_HELP_TEXT("Exit NxCamera")
+  },
+  {
+    "set",
+    "parameter d%",
+    nxcamera_cmd_set,
+    NXCAMERA_HELP_TEXT("Set camera control parameters")
+  }
+};
+
+static const char *controls[] =
+{
+  "brightness",
+  "contrast",
+  "saturation",
+  "hue",
+  "while_balance_automatic",
+  "exposure",
+  "gain_automatic",
+  "gain",
+  "horizontal_flip",
+  "vertical_flip",
+  "power_line_frequency",
+  "sharpness",
+  "auto_exposure"
+};
+
+static const int g_nxcamera_cmd_count = sizeof(g_nxcamera_cmds) /
+                                        sizeof(struct mp_cmd_s);
+
+/****************************************************************************
+ * Name: nxcamera_cmd_stream
+ *
+ *   nxcamera_cmd_loop() play and record the raw data file using the nxcamera
+ *   context.
+ *
+ ****************************************************************************/
+
+static int nxcamera_cmd_stream(FAR struct nxcamera_s *plooper, char *parg)
+{
+  int ret;
+  uint16_t width = 0;
+  uint16_t height = 0;
+  uint32_t framerate = 0;
+  uint32_t format = 0;
+  char cc[4] =
+    {
+      0
+    };
+
+  sscanf(parg, "%hd %hd %d %s", &width, &height, &framerate, cc);
+  format = v4l2_fourcc(cc[0], cc[1], cc[2], cc[3]);
+
+  /* Try to stream raw data with settings specified */
+
+  ret = nxcamera_stream(plooper, width, height, framerate, format);
+
+  /* nxcamera_stream returned values:
+   *
+   *   OK         Stream is being run
+   *   -EBUSY     The media device is busy
+   *   -ENOSYS    The video format is not unsupported
+   *   -ENODEV    No video device suitable to capture media
+   */
+
+  switch (-ret)
```

Comment:
That is ok, but personally I prefer `switch (ret)`

Reply:
sure